stack-graphs/src/filtered_stack_graph.rs

```diff
@@ -0,0 +1,386 @@
+use super::Filter;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+pub struct StackGraph {
+    files: Files,
+    nodes: Nodes,
+    edges: Edges,
+}
+
+impl StackGraph {
+    pub fn reconstruct(&self) -> crate::graph::StackGraph {
+        todo!()
+    }
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+#[serde(transparent)]
+pub struct Files {
+    data: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+#[serde(transparent)]
+pub struct Nodes {
+    data: Vec<Node>,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+#[serde(tag = "type")]
+pub enum Node {
+    #[serde(rename = "drop_scopes")]
```

Comment:
Would it be possible to use `#[serde(rename_all = "snake_case")]` on `Node`? It would save annotations and type opportunities. [Docs](https://serde.rs/container-attrs.html#rename_all).

Reply:
ah yes, completely forgot about that attribute.

this would serialize `JumpTo` as `jump_to`, whereas the [original implementation](https://github.com/github/stack-graphs/blob/2a75ad4d6e69441a0a9e05888947f47957decaea/stack-graphs/src/json.rs#L355) uses `jump_to_scope`. 
- shall we deviate from the original and use `jump_to` instead? or
- shall we keep the original nomenclature, and use a combination of `rename_all = "snake_case"` + `rename = "jump_to_scope"`?

Reply:
Just rename the enum case to JumpToScope, I'd say.